src/librustc_codegen_llvm/back/write.rs

```diff
@@ -1331,6 +1335,11 @@ fn execute_work_item(cgcx: &CodegenContext,
             let needs_lto = match cgcx.lto {
                 Lto::No => false,
 
+                // If the linker does LTO, we don't have to do it. Note that we
+                // keep doing full LTO, if it is requested, as not to break the
+                // assumption that the output will be a single module.
```

Comment:
To confirm, this is a rustc bug, right? This is fixable on our end where fat LTO plus cross-lang-lto shouldn't actually run the LTO passes in rustc?

Reply:
Yes, it probably isn't too hard to do just the module merging but not the optimizations. Maybe in another PR, unless you think it's urgent.

Reply:
Oh nah definitely fine to happen later, just wanted to make sure I understood!